src/vm/errors/vm_errors.rs

```diff
@@ -150,9 +155,22 @@ impl fmt::Display for VirtualMachineError {
             VirtualMachineError::AssertNotZero(value, prime) => {
                 write!(f, "Assertion failed, {} % {} is equal to 0", value, prime)
             },
+            VirtualMachineError::CantCreateDictionaryOnTakenSegment(index) => {
+                write!(f, "Error: Tried to create a dictionary on segment: {:?} when this segment already corresponds to a dictionary", index)
```

Comment:
Can't the segment belong to anything else? Is it being taken a guarantee that a dict took it? Essentially, could the message be misleading?

Reply:
Yes, this error is only produced by the DictManager when adding a tracker for the segment, so it being taken can only mean a dictionary was created on that segment. As it is an internal error (it shouldn't reach the user), I can make the error message a bit more clear, by making it more specific.